vars/uploadArtifacts.groovy

```diff
@@ -6,7 +6,7 @@
  * compatible open source license.
  */
 void call(Map args = [:]) {
-    def lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+    def lib = library(identifier: 'jenkins@1.0.0', retriever: legacySCM(scm))
```

Comment:
Can you please check if `@` is of any relevance for local files/libraries in jenkins? You can find more information here: https://www.jenkins.io/doc/book/pipeline/shared-libraries/#legacy-scm

Reply:
Thanks for the comment, @gaiksaya .. if you don't mind I put 'jenkins@main' for shared libs under /vars, so, it states the use of main branch clearly. Then, if anyone wants to override it, they can put override in each individual jenkinsfile as your comment at https://github.com/opensearch-project/opensearch-build-libraries/issues/7#issuecomment-1259999902

Reply:
Thats for jenkinsFile. However for libraries under `vars` we want it to be the same branch. For example, if the user is using `2.0.0` of this repository the libraries should check out `2.0.0` [here](https://github.com/opensearch-project/opensearch-build-libraries/pull/6/files#diff-4fc12a8f236ed19ed2a263445f244284dc46fdbc62576463d1bd26ba05c9de8fR10) and not main which might point to 3.0 and have a breaking change.

```
lib = library(identifier: 'jenkins@2.0.0', retriever: legacySCM(scm))
```
One way I can think storing that value in `version.properties` and loading it or need to check if above `legacySCM` has better way to checkout current commit than branches.

Reply:
Crystal clear to me now, makes sense and let me check and update. Thanks again @gaiksaya .